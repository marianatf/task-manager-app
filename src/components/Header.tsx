import React from "react";
import { CreateTask } from "./CreateTask";
import type { Priority } from "../types";

interface Props {
  onCreateTask: (title: string, description: string, priority: Priority) => void;
}

export const Header: React.FC<Props> = ({ onCreateTask }) => {
  return (
    <header>
      <h1 className="text-4xl font-bold text-center my-4">To-Do List</h1>
      <CreateTask addTask={onCreateTask} />
    </header>
  );
};
