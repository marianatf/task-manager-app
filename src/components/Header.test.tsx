import { Header } from "./Header";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

describe("Header", () => {
  const mockOnCreateTask = vi.fn();

  it("should match snapshot", () => {
    const { asFragment } = render(<Header onCreateTask={mockOnCreateTask} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
