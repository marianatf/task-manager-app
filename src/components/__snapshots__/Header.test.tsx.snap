// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="text-4xl font-bold text-center my-4"
    >
      To-Do List
    </h1>
    <form
      class="flex flex-col gap-3 m-4"
    >
      <input
        class="rounded-lg px-4 py-2 border border-gray-300 focus:outline-none focus:ring-2 focus:ring-black"
        placeholder="Add a task"
        value=""
      />
    </form>
  </header>
</DocumentFragment>
`;

exports[`Header > should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="text-4xl font-bold text-center my-4"
    >
      To-Do List
    </h1>
    <form
      class="flex flex-col gap-3 m-4"
    >
      <input
        class="rounded-lg px-4 py-2 border border-gray-300 focus:outline-none focus:ring-2 focus:ring-black"
        placeholder="Add a task"
        value=""
      />
    </form>
  </header>
</DocumentFragment>
`;
